src/olympia/files/tests/test_utils_.py

```diff
@@ -512,64 +512,73 @@ def test_extract_translations_fail_silent_missing_file(read_mock, file_obj):
 
-def test_resolve_i18n_message_uses_default_locale():
-    messages = {
-        'en-US': {
-            'foo': {'message': 'bar'}
+    def test_uses_default_locale(self):
+        messages = {
+            'en-US': {
+                'foo': {'message': 'bar'}
+            }
         }
-    }
-
-    result = utils.resolve_i18n_message('__MSG_foo__', messages, 'de', 'en')
-    assert result == 'bar'
 
+        result = utils.resolve_i18n_message(
+            '__MSG_foo__', messages, 'de', 'en')
+        assert result == 'bar'
 
-def test_resolve_i18n_message_no_locale_match():
-    # Neither `locale` or `locale` are found, "message" is returned unchanged
-    messages = {
-        'fr': {
-            'foo': {'message': 'bar'}
+    def test_no_locale_match(self):
+        # Neither `locale` or `locale` are found, "message" is returned
+        # unchanged
+        messages = {
+            'fr': {
+                'foo': {'message': 'bar'}
+            }
         }
-    }
 
-    result = utils.resolve_i18n_message('__MSG_foo__', messages, 'de', 'en')
-    assert result == '__MSG_foo__'
+        result = utils.resolve_i18n_message(
+            '__MSG_foo__', messages, 'de', 'en')
+        assert result == '__MSG_foo__'
 
+    def test_field_not_set(self):
+        """Make sure we don't fail on messages that are `None`
 
-def test_resolve_i18n_message_field_not_set():
-    """Make sure we don't fail on messages that are `None`
-
-    Fixes https://github.com/mozilla/addons-server/issues/3067
-    """
-    result = utils.resolve_i18n_message(None, {}, 'de', 'en')
-    assert result is None
-
-
-def test_resolve_i18n_message_field_no_string():
-    """Make sure we don't fail on messages that are no strings"""
-    result = utils.resolve_i18n_message([], {}, 'de', 'en')
-    assert result == []
+        Fixes https://github.com/mozilla/addons-server/issues/3067
+        """
+        result = utils.resolve_i18n_message(None, {}, 'de', 'en')
+        assert result is None
+
+    def test_field_no_string(self):
+        """Make sure we don't fail on messages that are no strings"""
+        result = utils.resolve_i18n_message([], {}, 'de', 'en')
+        assert result == []
+
+    def test_corrects_locales(self):
+        messages = {
+            'en-US': {
+                'foo': {'message': 'bar'}
+            }
+        }
 
+        result = utils.resolve_i18n_message('__MSG_foo__', messages, 'en')
+        assert result == 'bar'
 
-def test_resolve_i18n_message_corrects_locales():
-    messages = {
-        'en-US': {
-            'foo': {'message': 'bar'}
+    def test_ignore_wrong_format(self):
+        messages = {
+            'en-US': {
+                'foo': 'bar'
+            }
         }
-    }
 
-    result = utils.resolve_i18n_message('__MSG_foo__', messages, 'en')
-    assert result == 'bar'
+        result = utils.resolve_i18n_message('__MSG_foo__', messages, 'en')
+        assert result == '__MSG_foo__'
```

Comment:
Apart from the last method, correct.


On Tue, Sep 13, 2016, at 04:36 PM, Andrew Williamson wrote:
> In src/olympia/files/tests/test_utils_.py[1]:


> > > -    result = utils.resolve_i18n_message('__MSG_foo__', messages,
> > >      'en') > -    assert result == 'bar' > +        result =
> > >      utils.resolve_i18n_message('__MSG_foo__', messages, 'en') > +
> > >      assert result == '__MSG_foo__'
>
> Is this just refactoring?  Apart from the last line?
> — You are receiving this because you authored the thread. Reply to
> this email directly, view it on GitHub[2], or mute the thread[3].
>

Links:

  1. https://github.com/mozilla/addons-server/pull/3486#discussion_r78572287
  2. https://github.com/mozilla/addons-server/pull/3486/files/5e0f9b3d4508b56fa5d650a1b571910386a0a1f4#r78572287
  3. https://github.com/notifications/unsubscribe-auth/AAIfGVwe9QLN-3NYzn5Mic7gCpy6xc6Fks5qprTigaJpZM4J7eTk